pkg/restapi/operation/operations.go

```diff
@@ -204,19 +142,57 @@ func New(config *Config) (*Operation, error) {
 	return svc, nil
 }
 
-func openBootstrapStore(provider storage.Provider) (storage.Store, error) {
-	err := provider.CreateStore(bootstrapStoreName)
-	if err == nil {
-		logger.Infof(fmt.Sprintf("Created %s store.", bootstrapStoreName))
-	} else {
-		if !errors.Is(err, storage.ErrDuplicateStore) {
-			return nil, err
-		}
+// GetRESTHandlers get all controller API handler available for this service.
+func (c *Operation) GetRESTHandlers() []common.Handler {
+	return []common.Handler{
+		support.NewHTTPHandler(hydraLoginPath, http.MethodGet, c.hydraLoginHandler),
+		support.NewHTTPHandler(oauth2GetRequestPath, http.MethodGet, c.createOIDCRequest),
+		support.NewHTTPHandler(oauth2CallbackPath, http.MethodGet, c.handleOIDCCallback),
+		support.NewHTTPHandler(bootstrapGetRequestPath, http.MethodGet, c.handleBootstrapDataRequest),
+	}
+}
 
-		logger.Infof(fmt.Sprintf("%s store already exists. Skipping creation.", bootstrapStoreName))
+func (c *Operation) hydraLoginHandler(w http.ResponseWriter, r *http.Request) {
+	logger.Debugf("handling login request: %s", r.URL.String())
+
+	challenge := r.URL.Query().Get("login_challenge")
+	if challenge == "" {
+		common.WriteErrorResponsef(w, logger, http.StatusBadRequest, "missing challenge on login request")
+
+		return
 	}
 
-	return provider.OpenStore(bootstrapStoreName)
+	req := admin.NewGetLoginRequestParams()
+
+	req.SetLoginChallenge(challenge)
+
+	login, err := c.hydra.GetLoginRequest(req)
+	if err != nil {
+		common.WriteErrorResponsef(w, logger,
+			http.StatusBadGateway, "failed to fetch login request from hydra: %s", err.Error())
```

Comment:
Is http.StatusBadGateway the appropriate header for all errors returned from GetLoginRequest?

Reply:
Good question. I don't know. There is a `login.Error() string` method... I guess we can figure that out later.